vortex-scalar/src/primitive.rs

```diff
@@ -274,17 +274,51 @@ impl From<usize> for Scalar {
 }
 
 #[derive(Debug, Clone, Copy, PartialEq, Eq)]
+/// Binary element-wise operations on two arrays or two scalars.
 pub enum BinaryNumericOperator {
+    /// Binary element-wise addition of two arrays or of two scalars.
     Add,
+    /// Binary element-wise subtraction of two arrays or of two scalars.
     Sub,
+    /// Same as [BinaryNumericOperator::Sub] but with the parameters flipped: `right - left`.
+    FlippedSub,
```

Comment:
I'm not sure about these operators... it sort of forces everyone to handle bi-directional operators?

Reply:
If an array has no clever binary numeric operation, it just uses the Arrow implementation, but, yes, if an array implements subtract and it wants to handle constant right-hand-sides, it needs to deal with FlippedSub.

I suppose, if we had unary negation, we could use Add to flip. It seems bad to introduce a second pass over the array though.

An array _could_ implement just `Sub` and fallback to Arrow for `FlippedSub`.

I kinda think this is just the essential complexity of a non-commutative operation? Or maybe I misunderstand what you mean about bi-directional operators?

Reply:
This idea sounds cool but I think the clever binary numeric functions are easy to implemented for both LHS and RHS. Maybe other operators would have some benefits by doing this? I'm not sure.